data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+
+### Functions
+
+The type of functions from type `a` to type `b` is written `a -> b`. Here are a few examples:
+```ocaml
+# fun x -> x * x;;
+- : int -> int = <fun>
+
+# (fun x -> x * x) 9;;
+- : int = 81
+```
+
+The first expression is an anonymous function of type `int -> int`. The type is inferred from the expression `x * x`, which must be of type `int` since `*` is an operator that returns an `int`. The `<fun>` printed in place of the value is a token, meaning functions don't have a value to be displayed. This is because, if they have been compiled, their code is no longer available.
+
+The second expression is function application. The parameter `9` is applied, and the result `81` is returned.
+
+```ocaml
+# fun x -> x;;
+- : 'a -> 'a = <fun>
+
+# (fun x -> x) 42;;
+- : int = 42
+
+# (fun x -> x) "This is really disco!";;
+- : string = "This is really disco!"
+```
+
+The first expression is another anonymous function. It is the _identity_ function, it can be applied to anything, and it returns its argument unchanged. This means that its parameter can be of any type, and its result has the same type. The same code can be applied to data of different types. This is called _polymorphism_.
+
+Remember, the `'a` is a _type variable_, so values of any type can be passed to the function and replaces the type variable. The identity function has the same input and output type, whatever it may be.
+
+The two following expressions show that the identity function can apply to different type parameters:
+
+```ocaml
+# let f = fun x -> x * x;;
+val f : int -> int = <fun>
+
+# f 9;;
+- : int = 81
+```
+
+Defining a function is the same as naming a value, as illustrated in the first expression:
+
+```ocaml
+# let g x = x * x;;
+val g : int -> int = <fun>
+
+# g 9;;
+- : int = 81
+```
+
+Executable OCaml code consists primarily of functions, so it's beneficial to make them as concise and clear as possible. The function `g` is defined here using a shorter, more common, and maybe more intuitive syntax.
+
+In OCaml, functions may terminate without returning the expected type value by throwing an exception (`exn`), which does not appear in its type. There is no way to know if a function may raise an exception without inspecting its code.
+```ocaml
+# raise;;
+- : exn -> 'a' = <fun>
+```
+
+Exceptions are discussed in the [Error Handling](/docs/error-handling) guide.
+
+Functions may have several parameters.
+```ocaml
+# fun a b -> a ^ " " ^ b;;
+- : string -> string -> string = <fun>
+
+# let mean a b = (a + b) / 2;;
+val mean : int -> int -> int = <fun>
+```
+
+Like the product type symbol `*`, the function type symbol `->` is not associative. The following two types are not the same:
+- `(int -> int) -> int` : This function takes a function of type `int -> int` as a parameter and returns an `int` as the result.
+- `int -> (int -> int)` : This function takes an `int` asa  parameter and returns a function of type `int -> int` as the result.
```

Comment:
Is this similar to what we covered in Functions & Variables? About how `->` associates to the right? (li 562)

Reply:
Yes it is.